src/lro/lroPoller.ts

```diff
@@ -68,3 +97,48 @@ export class LROPoller<TResult extends BaseResult> extends Poller<
     return delay(this.intervalInMs);
   }
 }
+
+/**
+ * This function determines which strategy to use based on the response from
+ * the last operation executed, this last operation can be an initial operation
+ * or a polling operation. The 3 possible strategies are described below:
+ *
+ * A) Azure-AsyncOperation or Operation-Location
+ * B) Location
+ * C) BodyPolling (provisioningState)
+ *  - This strategy is used when:
+ *    - Response doesn't contain any of the following headers Location, Azure-AsyncOperation or Operation-Location
+ *    - Last operation method is PUT
+ */
+function getPollingStrategy<TResult extends BaseResult>(
+  initialOperation: LROOperationStep<TResult>,
+  sendOperationFn: SendOperationFn<TResult>,
+  finalStateVia?: FinalStateVia
+) {
+  const lroData = initialOperation.result._lroData;
+
+  if (!lroData) {
+    throw new Error(
+      "Expected lroData to be present for PollingStrategy discovery"
```

Comment:
This error eventually gets surfaced to the customer using our generated SDK right? I wonder if this should be a `RestError` then so we can include the request/response on the error for better troubleshooting.

Reply:
Make sense, made this a RestError, thanks for the suggestion